CMakeLists.txt

```diff
@@ -179,12 +179,17 @@ configure_file (
 
 # Only support GNU compilers at this time
 if (PIO_ENABLE_COVERAGE)
-  if (CMAKE_C_COMPILER_NAME STREQUAL "GNU")
```

Comment:
Is CMAKE_C_COMPILER_NAME (that we had before the change) valid/defined?

Reply:
CMAKE_C_COMPILER_NAME is valid/defined, however,
CMAKE_CXX_COMPILER_NAME is not.
CMAKE_<LANG>_COMPILER_ID is a compiler identification
string that is defined for C, CXX and Fortran so we can use
it instead.

Reply:
ok, thanks